### ISSUE TITLE
Ring/Pie improvements

### ISSUE THREAD

**Issue by AAAYaKo** [TLabAltoh/Unity-SDF-UI-Toolkit#22]
For pie and ring
Add start angle property
For reasons of similarity to figma, layout and animation
smth like this:
https://www.shadertoy.com/view/tlSGzG

![image](https://github.com/user-attachments/assets/5e6c2145-8773-4344-a4e5-fb719168fbbb)

Remove sdf arc, instead add corners rounding property to sdf ring(and rename sdf ring to arc)

https://github.com/user-attachments/assets/50fb3584-42e5-4dc2-839d-93b127ed7d32

**Comment by AAAYaKo** [comment]
https://github.com/user-attachments/assets/e6b344ad-e5f4-40ea-8726-9e105ac8ed33

**Comment by TLabAltoh** [comment]
I am considering how to render arc shape with corner rounding option like above video. But in my knowlage, maybe it needs spline to round arc's corners (but not sure).

<img src="https://github.com/user-attachments/assets/6cf3d7b1-5ff9-4825-9715-30f74ab24326" width="256"></img>

I don't want to use the spline approach for simple shapes because the distance function for splines is the heaviest and easily affects performance. So currently this update is not exactly the same as the video above.

<video src="https://github.com/user-attachments/assets/41126d73-27c3-48c4-96e0-db6947132734" width="256"></video>

<video src="https://github.com/user-attachments/assets/8fce19e1-62a2-4942-9495-d218548df886" width="256"></video>

<img src="https://github.com/user-attachments/assets/b4735fa8-d5fd-4605-aa33-91246af45512" width="256"></img>

**Comment by AAAYaKo** [comment]
![Screenshot_20240925_091335_Edge](https://github.com/user-attachments/assets/599d103a-fdc8-4785-9fd0-d5e176ee5d39)
No, I thought it could be done as mentioned in the article, if you have to use a spline for that then it's better not to do it.

**Comment by TLabAltoh** [comment]
About my latest update, I give up to implement exactly same as video you attached. So spline is not used, it is just implemented using round function as you mentioned.

**Comment by AAAYaKo** [comment]
I changed Arc settings a bit to be similar to figma settings. Also in its current form, there is no need for Pie anymore. There's probably no need for Circle too, but for performance and compatibility reasons I'd keep it. Maybe it makes sense to make one Circle script like in figma and swap shaders.
fill = 1 or -1 and ratio = 1 => circle
ratio = 1 => pie
otherwise => arc

**Comment by AAAYaKo** [comment]
What's changed:
Start angle is now measured in degrees
Width in pixels has been replaced by the ratio of the empty part to the filled part
Fill is now bidirectional
Rounding now measured in pixels
Start angle 0 is conformed to figma and math standards

**Comment by AAAYaKo** [comment]
https://github.com/user-attachments/assets/7d08e076-3d85-4721-9324-22da8d6d6273

**Comment by TLabAltoh** [comment]
I agree with the use of shader swaps for optimisation. And when considering the integration of circle based shapes into one. Maybe it is worth inheriting some ```SDFUI``` classes from each other like below because I want to use optimization with shader swap not only for ```Arc``` but also for ```Pie``` and others (like ```CutDisk```) to reduce ```if else``` statement from shader.

```cs
public class SDFArc: SDFPie {}
public class SDFPie: SDFCircle {}
public class SDFCutDisk: SDFCircle {}
```

I created a [```new branch```](https://github.com/TLabAltoh/Unity-SDF-UI-Toolkit/tree/shader-swap) and put my current approach in there for now (Currently, this branch is not intended to merge into the master immediately, so the shader is not modified and the if else statement still remains but If you are interested, I would appreciate any opinions or new commits for this branch).
